src/components/SchedulePanel/components/CourseSchedule/index.js

```diff
@@ -0,0 +1,144 @@
+import React, { Component, Fragment } from 'react';
+import Moment from 'moment';
+
+import Calendar from 'react-big-calendar/dist/react-big-calendar';
+import MomentLocalizer from 'react-big-calendar/lib/localizers/moment';
+
+import GordonLoader from '../../../../components/Loader';
+import schedule from './../../../../services/schedule';
+import myschedule from './../../../../services/myschedule';
+
+import './courseschedule.css';
+
+export default class CourseSchedule extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: true,
+      officeHoursOpen: false,
+      disabled: true,
+      selectedEvent: null,
+      isDoubleClick: false,
+    };
+    this.eventInfo = [];
```

Comment:
Is there a reason this isn't a part of the state?

Reply:
Because it contains data from backend and if we use setstate function, the promise order will be confused.